### PR TITLE
Add test_eventplot to test_datetime.py

### PR DESCRIPTION
**Pull request by shriyakalakata**
## PR summary

I have added a datetime smoke test/example code under the test_eventplot method in lib/matplotlib/test/test_datetime.py. This addresses the Axes.eventplot task from https://github.com/matplotlib/matplotlib/issues/26864. 

(edited) The plots generated from the latest test code:
eventplot

## PR checklist

- [x] "closes task Axes.eventplot in #26864" [link the related issue]
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [x] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: lib/matplotlib/tests/test_datetime.py [hunk 1]
```diff
@@ -260,11 +260,36 @@ def test_errorbar(self):
...
+        ], dtype=np.datetime64)
+
+        ax1.eventplot(x_dates1)
+
+        x_dates2 = np.array([
+            [datetime.datetime(2020, 6, 30), datetime.datetime(2020, 7, 22),
+             datetime.datetime(2020, 8, 3), datetime.datetime(2020, 9, 14)],
+            [datetime.datetime(2020, 7, 18), datetime.datetime(2020, 7, 21),
+             datetime.datetime(2020, 8, 3), datetime.datetime(2020, 10, 14)]
+        ], dtype=np.datetime64)
```
[the remark this answers is not in the document]
Comment (shriyakalakata) [reply lib/matplotlib/tests/test_datetime.py:283]: Good point. I'll modify it accordingly.

File: lib/matplotlib/tests/test_datetime.py [hunk 2]
```diff
@@ -260,11 +260,37 @@ def test_errorbar(self):
...
+                            dtype=np.datetime64,
+                            )
+
+        ax1.eventplot(x_dates1)
+
+        start_date = datetime.datetime(2020, 7, 1)
+        end_date = datetime.datetime(2020, 10, 15)
+        date_range = end_date - start_date
+
+        dates1 = start_date + np.random.rand(30) * date_range
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_datetime.py:282]: If you want to use randomness, then you will need to set a seed (we use 19680801 for tests), or rewrite without randomness.

**Review by QuLogic — COMMENTED** [review]
On a side note, you should probably [set your email address in your git config](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address). The first commit is using a random DHCP-supplied hostname from NYU, and the second is using your default macbook hostname.

File: lib/matplotlib/tests/test_datetime.py (hunk 2 above)
Comment (shriyakalakata) [reply lib/matplotlib/tests/test_datetime.py:284]: Fixed both!

**Comment by shriyakalakata** [comment]
> I think we may wish to add a testcase which exercise the units behavior on the `y` axis (i.e. `line offset` and `linelength` params, set to `datetime` and `timedelta`, respectively)
> 
I'm able to use `datetime` for `lineoffset` on the y axis but seems like `linelength` can't be of type `timedelta`

**Comment by QuLogic** [comment]
> I'm able to use `datetime` for `lineoffset` on the y axis but seems like `linelength` can't be of type `timedelta`

I've opened #27479 to track this.

**Review by ksunden — APPROVED** [review]
Since `linelengths` is not well supported (and we have an issue to track getting it supported) I'll go ahead and approve this for what it is. We can edit it to include `linelengths` as timedeltas when that has been addressed.

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
Thanks @shriyakalakata! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Pull request merged**